### ISSUE TITLE
error after 9.0.0 addon update

### ISSUE THREAD

**Issue by Daeveren** [Jaliborc/PetTracker#221]
**Running Software (issues missing this information will be deleted):**
 - Addon version: 9.0.0
 - Server patch: [e.g. Build 8.0.1.27178] 9.0

 **Have you read the changelog? (please don't waste our time)**
 Yes

**Describe the bug**
Getting this error even after the 9.0 addon update

```
Message: ...ce\AddOns\PetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)
Time: Thu Oct 15 14:07:46 2020
Count: 3
Stack: ...ce\AddOns\PetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)
[string "@Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua"]:758: in function <...Ons\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:717>
[string "=[C]"]: ?
[string "@Interface\AddOns\PetTracker\addons\main\features\gossip.lua"]:39: in function <...ce\AddOns\PetTracker\addons\main\features\gossip.lua:38>
[string "=[C]"]: in function `SelectOption'
[string "@Interface\FrameXML\GossipFrame.lua"]:74: in function `GossipFrame_HandleShow'
[string "@Interface\FrameXML\CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
[string "@Interface\FrameXML\CustomGossipFrameBase.lua"]:39: in function <Interface\FrameXML\CustomGossipFrameBase.lua:37>

Locals: errMsg = "...ce\AddOns\PetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)"
isBugGrabber = nil
tsmErrMsg = nil
oldModule = nil
private = <table> {
 globalNameTranslation = <table> {
 }
 FormatErrorMessageSection = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:517
 ErrorHandler = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:134
 origErrorHandler = <function> defined @Interface\SharedXML\SharedBasicControls.lua:355
 hitInternalError = false
 CreateErrorFrame = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:536
 num = 0
 SanitizeString = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:510
 AddonBlockedHandler = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:502
 GetStackLevelInfo = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:344
 ParseLocals = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:396
 ignoreErrors = false
 errorReports = <table> {
 }
 IsTSMAddon = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:478
 localLinesTemp = <table> {
 }
 GetStackInfo = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:313
}
MAX_STACK_DEPTH = 50
OLD_TSM_MODULES = <table> {
 1 = "TradeSkillMaster_Accounting"
 2 = "TradeSkillMaster_AuctionDB"
 3 = "TradeSkillMaster_Auctioning"
 4 = "TradeSkillMaster_Crafting"
 5 = "TradeSkillMaster_Destroying"
 6 = "TradeSkillMaster_Mailing"
 7 = "TradeSkillMaster_Shopping"
 8 = "TradeSkillMaster_Vendoring"
 9 = "TradeSkillMaster_Warehousing"
}
```

**Comment by Palonblade** [comment]
I am still getting this error or similar at least:

1x ...cePetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)
[string "@PetTracker\addons\main\features\gossip.lua"]:39: in function <...cePetTracker\addons\main\features\gossip.lua:38>
[string "=[C]"]: in function `SelectOption'
[string "@FrameXML\GossipFrame.lua"]:74: in function `GossipFrame_HandleShow'
[string "@FrameXML\CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
[string "@FrameXML\CustomGossipFrameBase.lua"]:39: in function <FrameXML\CustomGossipFrameBase.lua:37>

Locals:
index = 1
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'buttons' (a nil value)"
ADDON = "PetTracker"
Addon = <table> {
 SendSignal = <function> defined @PetTracker\libs\WildAddon-1.0\WildAddon-1.0.lua:82
 Gossip = <table> {
 }
 SpeciePin = <table> {
 }
 Tutorials = <table> {
 }
 UnregisterAllEvents = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 Options = <unnamed> {
 }
 Enemy = <table> {
 }
 sets = <table> {
 }
 state = <table> {
 }
 MaxLevel = 25
 GetColor = <function> defined @PetTracker\addons\main\main.lua:66
 Entity = <table> {
 }
 Species = <table> {
 }
 Delaying = <function> defined @PetTracker\libs\MutexDelay-1.0\MutexDelay-1.0.lua:46
 JournalSlot = <table> {
 }
 Rival = <table> {
 }
 Tooltips = <table> {
 }
 OnPetsChanged = <function> defined @PetTracker\addons\main\main.lua:53
 NewModule = <function> defined @PetTracker\libs\WildAddon-1.0\WildAddon-1.0.lua:68
 MultiTip = <table> {
 }
 MapCanvas = <table> {
 }
 Tracker = <table> {
 }
 Stables = <table> {
 }
 AbilityDisplay = <table> {
 }
 TextLine = <table> {
 }
 Search = <function> defined @PetTracker\addons\main\api\search.lua:26
 StablePin = <table> {
 }
 RivalInfo = <table> {
 }
 Pin = <table> {
 }
 Delay = <function> defined @PetTracker\libs\MutexDelay-1.0\MutexDelay-1.0.lua:27
 Tag = "PETTRACKER_"
 BattleSlot = <table> {
 }
 Pet = <table> {
 }
 AbilityAction = <table> {
 }
 TextList = <table> {
 }
 MaxQuality = 6
 UnregisterMessage = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 ProgressBar = <table> {
 }
 SpecieStats = <table> {
 }
 RivalPin = <table> {
 }
 MapSearch = <table> {
 }
 Specie = <table> {
 }
 PetSlot = <table> {
 }
 Ability = <table> {
 }
 Rivals = <table> {
 }
 Base = <table> {
 }
 Predict = <table> {
 }
 UnregisterSignal = <function> defined @PetTracker\libs\WildAddon-1.0\WildAddon-1.0.lua:78
 Maps = <table> {
 }
 SpecieBreeds = <table> {
 }
 RegisterMessage = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 RegisterEvent = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 UnregisterAllMessages = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 OnBattle = <function> defined @PetTracker\addons\main\main.lua:57
 Breeds = <table> {
 }
 AbilityButton = <table> {
 }
 OnEnable = <function> defined @PetTracker\addons\main\main.lua:26
 Name = "PetTracker"
 RivalOrder = <table> {
 }
 SendMessage = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:59
 RegisterSignal = <function> defined @PetTracker\libs\WildAddon-1.0\WildAddon-1.0.lua:74
 UnregisterEvent = <function> defined @Armory\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 Objectives = <unnamed> {
 }
}

**Comment by alexnaumovich** [comment]
Why was this issue closed? It is still happening with the 9.0.0 update.

- Login
- Go to Boralis
- Click on the mission table

Message: ...ce\AddOns\PetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)
Time: Thu Oct 15 23:21:45 2020
Count: 2
Stack: ...ce\AddOns\PetTracker\addons\main\features\gossip.lua:39: attempt to index field 'buttons' (a nil value)
[string "@Interface\AddOns\PetTracker\addons\main\features\gossip.lua"]:39: in function <...ce\AddOns\PetTracker\addons\main\features\gossip.lua:38>
[string "=[C]"]: in function `SelectOption'
[string "@Interface\FrameXML\GossipFrame.lua"]:74: in function `GossipFrame_HandleShow'
[string "@Interface\FrameXML\CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
[string "@Interface\FrameXML\CustomGossipFrameBase.lua"]:39: in function <Interface\FrameXML\CustomGossipFrameBase.lua:37>

Locals: index = 1
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'buttons' (a nil value)"
ADDON = "PetTracker"
Addon = <table> {
 SendSignal = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\WildAddon-1.0\WildAddon-1.0.lua:82
 Gossip = <table> {
 }
 SpeciePin = <table> {
 }
 UnregisterAllEvents = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 Enemy = <table> {
 }
 sets = <table> {
 }
 state = <table> {
 }
 MaxLevel = 25
 GetColor = <function> defined @Interface\AddOns\PetTracker\addons\main\main.lua:66
 Entity = <table> {
 }
 Species = <table> {
 }
 Delaying = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\MutexDelay-1.0\MutexDelay-1.0.lua:46
 JournalSlot = <table> {
 }
 Rival = <table> {
 }
 Tooltips = <table> {
 }
 OnPetsChanged = <function> defined @Interface\AddOns\PetTracker\addons\main\main.lua:53
 NewModule = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\WildAddon-1.0\WildAddon-1.0.lua:68
 MultiTip = <table> {
 }
 MapCanvas = <table> {
 }
 Tracker = <table> {
 }
 Stables = <table> {
 }
 AbilityDisplay = <table> {
 }
 TextLine = <table> {
 }
 Search = <function> defined @Interface\AddOns\PetTracker\addons\main\api\search.lua:26
 StablePin = <table> {
 }
 RivalInfo = <table> {
 }
 Pin = <table> {
 }
 Delay = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\MutexDelay-1.0\MutexDelay-1.0.lua:27
 Tag = "PETTRACKER_"
 BattleSlot = <table> {
 }
 Pet = <table> {
 }
 AbilityAction = <table> {
 }
 TextList = <table> {
 }
 MaxQuality = 6
 UnregisterMessage = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 ProgressBar = <table> {
 }
 SpecieStats = <table> {
 }
 RivalPin = <table> {
 }
 MapSearch = <table> {
 }
 Specie = <table> {
 }
 PetSlot = <table> {
 }
 Ability = <table> {
 }
 Rivals = <table> {
 }
 Base = <table> {
 }
 Predict = <table> {
 }
 UnregisterSignal = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\WildAddon-1.0\WildAddon-1.0.lua:78
 Maps = <table> {
 }
 SpecieBreeds = <table> {
 }
 RegisterMessage = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 RegisterEvent = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 UnregisterAllMessages = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 OnBattle = <function> defined @Interface\AddOns\PetTracker\addons\main\main.lua:57
 Breeds = <table> {
 }
 AbilityButton = <table> {
 }
 OnEnable = <function> defined @Interface\AddOns\PetTracker\addons\main\main.lua:26
 Name = "PetTracker"
 RivalOrder = <table> {
 }
 SendMessage = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:59
 RegisterSignal = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\WildAddon-1.0\WildAddon-1.0.lua:74
 UnregisterEvent = <function> defined @Interface\AddOns\Bagnon\common\Wildpants\libs\CallbackHandler-1.0\CallbackHan
